app/pages/ServiceDiscoveryForm/ServiceDiscoveryForm.tsx

```diff
@@ -4,63 +4,117 @@ import qs from 'qs';
+
+  // Add the selectedRadioItem to state so that when a user clicks the next button
+  // the selectedRadioItem can be passed to the goToNextStep function
+  const [selectedRadioItem, setSelectedRadioItem] = useState<number | null>(null);
+
+  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(activeCategoryId)
+  || [];
+  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(activeCategoryId)
+  || [];
+
   return (
     <InnerServiceDiscoveryForm
-      categorySlug={category.slug}
+      category={category}
+      subcategorySubheading={category.subcategorySubheading}
       eligibilities={eligibilities}
       subcategories={subcategories}
-      steps={category.steps}
-      subcategorySubheading={category.subcategorySubheading}
+      setActiveCategoryId={setActiveCategoryId}
+      selectedRadioItem={selectedRadioItem}
+      setSelectedRadioItem={setSelectedRadioItem}
     />
   );
 };
 
 /** Main component that handles form data and advancing steps. */
 const InnerServiceDiscoveryForm = ({
-  steps, eligibilities, subcategories, categorySlug, subcategorySubheading,
+  category, subcategorySubheading, eligibilities, subcategories,
+  setActiveCategoryId, selectedRadioItem, setSelectedRadioItem,
 }: {
-  steps: Step[];
+  category: ServiceCategory;
+  subcategorySubheading: string;
   eligibilities: CategoryRefinement[];
   subcategories: CategoryRefinement[];
-  categorySlug: string;
-  subcategorySubheading: string;
+  setActiveCategoryId: (id: string | null) => void;
+  selectedRadioItem: number | null;
+  setSelectedRadioItem: (item: number | null) => void;
 }) => {
-  const [currentStep, setCurrentStep] = useState(0);
+  const {
+    steps,
+    slug: categorySlug,
+  } = category;
 
-  // TODO: Should goBack go back to the previous step?
+  const [currentStep, setCurrentStep] = useState(0);
   const history = useHistory();
-  const goBack = () => history.goBack();
-  const goToNextStep = () => setCurrentStep(currentStep + 1);
+  let goToNextStep;
+  const stepName = steps[currentStep];
+  const disableNextBtn = selectedRadioItem === null && ['housingStatus', 'subcategoriesRadio'].includes(stepName);
+  // TODO: Should goBack go back to the previous step?
+  const goBack = () => {
+    history.push('/');
+  };
+
+  if (stepName === 'housingStatus') {
+    // TODO: If we have more categories that have custom pathway actions, will we want to
+    // store custom methods as children of the categories in the constants.ts file?
+    goToNextStep = () => {
+      if (!selectedRadioItem) {
+        return;
+      }
+
+      if (selectedRadioItem === 1100045) {
+        // User has selected first option in Long Term Housing step 1. Set category ID to Shelter
+        // ID and redirect user to the Shelter resources form.
+        setActiveCategoryId('1000010');
+        history.replace('/shelter-resources/form');
+      } else {
+        const targetCategoryId = (selectedRadioItem);
```

Comment:
```suggestion
        const targetCategoryId = selectedRadioItem;
```

Nit: No parentheses needed here.